### PR TITLE
Feature/vuex gateway init

### LINKED ISSUE #22: Feature/Vuex uses AppGatewayClient for GRPC requests
## Description
The Vue.js's idiomatic way of managing state and http requests is through [vuex](https://vuex.vuejs.org/guide/actions.html) `actions` property.

I believe the recommended workflow is the following:
- vuex holds the global state, and it will have `getters`, `mutations`,` actions`.
- all http requests is performed in the `actions` property due to the following [reasons](https://vuex.vuejs.org/guide/actions.html):
    - Instead of mutating the state, actions commit mutations
    - `actions` can contain arbitrary asynchronous operations
    - The only way to actually change state in a Vuex store is by committing a mutation in the `mutations` property.

Examples: https://vuex.vuejs.org/guide/actions.html

## Definitions of Done
- [x] a modularized `app-gateway-svc` client if still needed
- [x] ability of a component to dispatch an action. an example would be https://vuex.vuejs.org/guide/actions.html#dispatching-actions used in https://github.com/vuejs/vuex/blob/dev/examples/shopping-cart/components/ShoppingCart.vue#L33 and performs all the mutations in its definition https://github.com/vuejs/vuex/blob/dev/examples/shopping-cart/store/modules/cart.js#L32
- [x] follows the workflow `dispatch` from vue component -> `commit` from vuex's action -> mutate the state from vuex's mutation
- [ ] unit tested

## Note
`app-gateway-svc` client is not the only client. We should be able to use different clients within Vuex's `mutations` property e.g. `elastic-search client` + `app-gateway-svc client` + etc.

### PR DESCRIPTION
**Pull request by kimlisa**
closes #22

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,12 +16,14 @@
   "dependencies": {
     "@hwsc/hwsc-api-blocks-lisa-test": "0.0.3",
     "core-js": "^3.0.1",
+    "grpc-web": "^1.0.4",
     "vue": "^2.6.10",
     "vue-class-component": "^7.0.2",
     "vue-i18n": "^8.0.0",
     "vue-property-decorator": "^8.1.0",
```
Comment (faraonc) [inline package.json]: 8.1.1 is out

File: src/store/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+# Store Guide
```
Comment (faraonc) [inline src/store/README.md:1]: We should reference this README.md in the project's parent directory README.md.

File: src/store/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+[Learn about Vuex State](https://vuex.vuejs.org/guide/state.html)
+
+### getters.ts
+TODO
+
+### actions.ts
+Contains exported defined action methods for the store module.
+
+The difference between `mutations` and `actions` is that `actions` contains the heavy logic,
```
Comment (faraonc) [inline src/store/README.md]: https://vuex.vuejs.org/guide/actions.html 
Not necessarily, better to rephrase this as `actions` is the business logic and can commit multiple `mutations` at a time. `mutations` may have logic dealing with mutating the state.

File: src/store/modules/app-gateway-svc/actions.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+export const setNewClient = ({ state, commit }: ActionContext): any => {
+  const address: string | undefined = process.env.VUE_APP_HOST_NAME;
+  const hostname: string = address == null ? '' : address;
+
+  if (hostname.length === 0) {
+    // TODO not sure what to do here
+    console.error('setNewClient: VUE_APP_HOST_NAME env var not loaded');
+  }
```
Comment (faraonc) [inline src/store/modules/app-gateway-svc/actions.ts]: I think we can use truthy here.
```ts
const hostname: string = process.env.VUE_APP_HOST_NAME;

if (!hostname) {
    // TODO not sure what to do here
    console.error('setNewClient: VUE_APP_HOST_NAME env var not loaded');
    // route to error 50X page
  }
```

Comment (faraonc) [reply src/store/modules/app-gateway-svc/actions.ts]: We can probably removed the alternate type `undefined` as well.

File: src/store/modules/app-gateway-svc/actions.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  // eslint-disable-next-line max-len
+  const client: AppGatewayServiceClient = new AppGatewayServiceClient(hostname, null, null);
+  commit(mutation.SET_GRPC_CLIENT, client);
+};
+
+export const setNewHttpHeader = ({ state, commit }: ActionContext): any => {
```
Comment (faraonc) [inline src/store/modules/app-gateway-svc/actions.ts]: I think this should be renamed to `setAuthHeader` because it is only dealing with authentication/authorization.

File: src/store/modules/app-gateway-svc/actions.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  let token: string | null = window.localStorage.getItem(consts.LOCAL_STORAGE_TOKEN_KEY);
+  let authType: string = consts.USER_AUTH;
+
+  if (token == null) {
+    token = window.btoa(
+      `${process.env.VUE_APP_DUMMY_EMAIL}:${process.env.VUE_APP_DUMMY_PASSWORD}`,
+    );
+    authType = consts.BASIC_AUTH;
+  }
+
```
Comment (faraonc) [inline src/store/modules/app-gateway-svc/actions.ts]: Minimize mutation and alternate types as much as possible.

I think this can work.
```ts
const isAuthToken: boolean = window.localStorage.getItem(consts.LOCAL_STORAGE_TOKEN_KEY)? true : false;

const token: string = isAuthToken? window.localStorage.getItem(consts.LOCAL_STORAGE_TOKEN_KEY) : window.btoa(
      `${process.env.VUE_APP_DUMMY_EMAIL}:${process.env.VUE_APP_DUMMY_PASSWORD}`,
    );

const authType: string  = isAuthToken?  consts.USER_AUTH :  consts.USER_AUTH
```

File: src/store/modules/app-gateway-svc/states.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import { ModuleState } from './types';
+
+// eslint-disable-next-line import/prefer-default-export
+export const state: ModuleState = {
+  grpcClient: undefined as any,
+  httpHeader: undefined as any,
```
Comment (faraonc) [inline src/store/modules/app-gateway-svc/states.ts]: why `undefined as any`? We should make them as optional properties with `?`

Comment (faraonc) [reply src/store/modules/app-gateway-svc/states.ts]: I do not understand how `states.ts` and `types.ts`. Why do we need both?

Comment (faraonc) [reply src/store/modules/app-gateway-svc/states.ts]: We can get rid of `types.ts`.

Why not just include the interface here to reduce the file to manage. The boiler plating is getting confusing.

```ts
import { AppGatewayServiceClient, grpc } from '../../root-types';

export interface ModuleState {
  grpcClient?: AppGatewayServiceClient;
  httpHeader?: grpc.Metadata;
}

export const state: ModuleState = {};
```

Comment (faraonc) [inline src/store/modules/app-gateway-svc/states.ts]: `httpHeader` is more like `authHeader` at the moment. If we have to change it to `httpHeader` later on, then we can do it later on. I think it is more specific and easier to understand.

File: src/store/modules/app-gateway-svc/mutations.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { MutationTree } from 'vuex';
+import { ModuleState } from './types';
+import * as mutation from './types-mutations';
+import { AppGatewayServiceClient, grpc } from '../../root-types';
+
+// eslint-disable-next-line import/prefer-default-export
+export const mutations: MutationTree<ModuleState> = {
+  [mutation.SET_HTTP_HEADER](state, payload: grpc.Metadata) {
+    Object.assign(state, { httpHeader: payload });
+    // state.httpHeader = payload;
```
Comment (faraonc) [inline src/store/modules/app-gateway-svc/mutations.ts]: We should probably rename the `state` param to maybe `newState` ?

File: src/store/modules/app-gateway-svc/index.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+import { state } from './states';
+import * as actions from './actions';
+import { mutations } from './mutations';
+import { ModuleState } from './types';
+import { RootState } from '../../root-types';
+
+// namespace name is set to the module name: appGateway
+const namespaced: boolean = true;
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (faraonc) [inline src/store/modules/app-gateway-svc/index.ts]: Maybe?
```ts
const appGateway: Module<ModuleState, RootState> = {
  namespaced,
  state,
  actions,
  mutations,
};

export default appGateway;
```

**Review by faraonc — COMMENTED** [review]
Added my thoughts for you to consider.

File: package.json (hunk 1 above)
Comment (kimlisa) [reply package.json]: updated

File: src/store/modules/app-gateway-svc/actions.ts (hunk 4 above)
Comment (kimlisa) [reply src/store/modules/app-gateway-svc/actions.ts]: it gives type error because the env var may not exist which is than "undefined" and we cant store undefined as strings.

File: src/store/modules/app-gateway-svc/actions.ts (hunk 5 above)
Comment (kimlisa) [reply src/store/modules/app-gateway-svc/actions.ts]: i renamed action to `initAuthHeader` and renamed mutation to `setAuthHeader`

File: src/store/modules/app-gateway-svc/actions.ts (hunk 6 above)
Comment (kimlisa) [reply src/store/modules/app-gateway-svc/actions.ts]: ```window.localStorage.getItem(consts.LOCAL_STORAGE_TOKEN_KEY)``` does not return a boolean, it returns type "string or null". Typescript is really strict

Comment (kimlisa) [reply src/store/modules/app-gateway-svc/actions.ts]: strangely, this works 
```
const isAuthToken: boolean = !!window.localStorage.getItem(consts.LOCAL_STORAGE_TOKEN_KEY);
``` 
Note the double exclamation mark. Not sure how that works right of the bat

Comment (kimlisa) [reply src/store/modules/app-gateway-svc/actions.ts]: Do you want me to use the double exclamation mark point?

File: src/store/modules/app-gateway-svc/states.ts (hunk 7 above)
Comment (kimlisa) [reply src/store/modules/app-gateway-svc/states.ts]: > why `undefined as any`? We should make them as optional properties with `?`

Because I have to initialize it with something and type null or undefined is not a valid type for the type I defined it as. So `undefined as any` works for any type.

I can't make them optional, it has to exist already for my IDE to recognize it because I make references to this state in actions or other files.

Comment (kimlisa) [reply src/store/modules/app-gateway-svc/states.ts]: > I do not understand how `states.ts` and `types.ts`. Why do we need both?

That's because this module is very small, so it might be hard to see the difference. This module's only purpose is to initialize client and http header when app.vue starts (atm).

`state.ts` only contain state of the module, `types.ts` includes any interfaces and other typings that the module will need.

Comment (kimlisa) [reply src/store/modules/app-gateway-svc/states.ts]: > We can get rid of `types.ts`.
> 
> Why not just include the interface here to reduce the file to manage. The boiler plating is getting confusing.
> 
> ```ts
> import { AppGatewayServiceClient, grpc } from '../../root-types';
> 
> export interface ModuleState {
>   grpcClient?: AppGatewayServiceClient;
>   httpHeader?: grpc.Metadata;
> }
> 
> export const state: ModuleState = {};
> ```

I was thinking of that too. The file could potentially get large, but I agree its confusing.

File: src/store/modules/app-gateway-svc/mutations.ts (hunk 8 above)
Comment (kimlisa) [reply src/store/modules/app-gateway-svc/mutations.ts]: It's either a convention to use `state` or it actually refers to the `export const state` object we defined in `states.ts`. I know in `actions.ts`, `state` definitely refers to the defined state object and can't change the name.

File: src/store/modules/app-gateway-svc/index.ts (hunk 9 above)
Comment (kimlisa) [reply src/store/modules/app-gateway-svc/index.ts]: I read that we should avoid using export defaults: https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html

File: src/store/modules/app-gateway-svc/actions.ts (hunk 6 above)
Comment (kimlisa) [reply src/store/modules/app-gateway-svc/actions.ts]: Actually, to enforce a boolean, i can do this

```
let isAuthToken: boolean = window.localStorage.getItem(consts.LOCAL_STORAGE_TOKEN_KEY) === null;
```

File: src/store/modules/app-gateway-svc/actions.ts (hunk 4 above)
Comment (faraonc) [reply src/store/modules/app-gateway-svc/actions.ts]:
```ts
 const hostname: string | undefined = process.env.VUE_APP_HOST_NAME;
   if (!hostname) {
    // TODO not sure what to do here
    console.error('setNewClient: VUE_APP_HOST_NAME env var not loaded');
  // route to error 500 page
  }
```

File: src/store/modules/app-gateway-svc/index.ts (hunk 9 above)
Comment (faraonc) [reply src/store/modules/app-gateway-svc/index.ts]: the eslint says something different. i think people have different opinions. if we are going to use a linter, might as well stick to it?

File: src/store/README.md (hunk 2 above)
Comment (faraonc) [reply src/store/README.md:1]: Is this resolved already?

File: src/store/README.md (hunk 3 above)
Comment (faraonc) [reply src/store/README.md]: Is this resolved or am I seeing old commits?

File: src/store/modules/app-gateway-svc/actions.ts (hunk 6 above)
Comment (faraonc) [reply src/store/modules/app-gateway-svc/actions.ts]: Try not to use `let` keyword as much as possible is the best practice.

File: src/store/modules/app-gateway-svc/states.ts (hunk 7 above)
Comment (faraonc) [reply src/store/modules/app-gateway-svc/states.ts]: If it is confusing then we should get rid of it?

File: src/store/modules/app-gateway-svc/actions.ts (hunk 4 above)
Comment (faraonc) [reply src/store/modules/app-gateway-svc/actions.ts]: ooppppps
```ts
const hostname: string = process.env.VUE_APP_HOST_NAME? '' : address;
 if (!hostname) {
    // TODO not sure what to do here
    console.error('setNewClient: VUE_APP_HOST_NAME env var not loaded');
  // route to error 500 page
  }
```

File: src/store/modules/app-gateway-svc/actions.ts (hunk 6 above)
Comment (kimlisa) [reply src/store/modules/app-gateway-svc/actions.ts]: I am open for more feedback after you take a look at the new block. I had to put in more cases.

File: src/store/modules/gateway/actions.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+interface ActionContext {
+  state: GatewayState;
+  rootState: RootState;
+  commit: Commit;
+  payload?: ActionPayload;
+}
+
+export const setNewClient = ({ state, commit }: ActionContext): any => {
+  const hostname: string = process.env.VUE_APP_HOST_NAME == null ? '' : process.env.VUE_APP_HOST_NAME;
```
Comment (faraonc) [inline src/store/modules/gateway/actions.ts]: Another JS feature, you prolly interested with is `||`.

`const hostname: string = process.env.VUE_APP_HOST_NAME || '';`

File: src/store/modules/gateway/actions.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+export const initAuthHeader = ({ state, commit }: ActionContext): any => {
+  let token: string | null = window.localStorage.getItem(consts.LOCAL_STORAGE_TOKEN_KEY);
+  let authType: string = consts.USER_AUTH;
+
+  if (token == null) {
+    const dummyEmail: string | undefined = process.env.VUE_APP_DUMMY_EMAIL;
+    const dummyPassword: string | undefined = process.env.VUE_APP_DUMMY_PASSWORD;
+    if (dummyEmail == null && dummyPassword == null) {
```
Comment (faraonc) [inline src/store/modules/gateway/actions.ts]: Did you mean `dummyEmail == null || dummyPassword == null`?

**Review by faraonc — CHANGES_REQUESTED** [review]

File: package.json [hunk 12]
```diff
@@ -16,12 +16,14 @@
   "dependencies": {
```
Comment (faraonc) [inline package.json:16]: running `npm run serve` gets this error.
```
ERROR in /Users/faraonc/WebstormProjects/hwsc-frontend/node_modules/@hwsc/hwsc-api-blocks-lisa-test/protobuf/hwsc-app-gateway-svc/app/hwsc-app-gateway-svc_pb.d.ts
1:23 Could not find a declaration file for module 'google-protobuf'. '/Users/faraonc/WebstormProjects/hwsc-frontend/node_modules/google-protobuf/google-protobuf.js' implicitly has an 'any' type.
  Try `npm install @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf';`
  > 1 | import * as jspb from "google-protobuf"
      |                       ^
    2 | 
    3 | import * as protobuf_hwsc$user$svc_user_hwsc$user$svc_pb from '../../../protobuf/hwsc-user-svc/user/hwsc-user-svc_pb';
    4 | import * as protobuf_lib_user_pb from '../../../protobuf/lib/user_pb';
```

Do we need ` npm install @types/google-protobuf`?

Comment (faraonc) [reply package.json:16]: `npm install @types/google-protobuf` seems to fix the issue but still
```
faraonc@~/WebstormProjects/hwsc-frontend (feature/vuex-gateway-init)$ npm run serve

> hwsc-frontend@0.0.1 serve /Users/faraonc/WebstormProjects/hwsc-frontend
> vue-cli-service serve

 INFO  Starting development server...
Starting type checking service...
Using 1 worker with 2048MB memory limit
 98% after emitting CopyPlugin                                                    

 WARNING  Compiled with 2 warnings                                                                                                                                                                                                                               12:55:36 AM

Module Warning (from ./node_modules/eslint-loader/index.js):
error: Line 11 exceeds the maximum line length of 120 (max-len) at src/components/HelloWorld.vue:11:1:
   9 |     <h3>Installed CLI Plugins</h3>
  10 |     <ul>
> 11 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel" target="_blank" rel="noopener">babel</a></li>
     | ^
  12 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-typescript" target="_blank" rel="noopener">typescript</a></li>
  13 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint" target="_blank" rel="noopener">eslint</a></li>
  14 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest" target="_blank" rel="noopener">unit-jest</a></li>


error: Line 12 exceeds the maximum line length of 120 (max-len) at src/components/HelloWorld.vue:12:1:
  10 |     <ul>
  11 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel" target="_blank" rel="noopener">babel</a></li>
> 12 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-typescript" target="_blank" rel="noopener">typescript</a></li>
     | ^
  13 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint" target="_blank" rel="noopener">eslint</a></li>
  14 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest" target="_blank" rel="noopener">unit-jest</a></li>
  15 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-e2e-cypress" target="_blank" rel="noopener">e2e-cypress</a></li>


error: Line 13 exceeds the maximum line length of 120 (max-len) at src/components/HelloWorld.vue:13:1:
  11 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel" target="_blank" rel="noopener">babel</a></li>
  12 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-typescript" target="_blank" rel="noopener">typescript</a></li>
> 13 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint" target="_blank" rel="noopener">eslint</a></li>
     | ^
  14 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest" target="_blank" rel="noopener">unit-jest</a></li>
  15 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-e2e-cypress" target="_blank" rel="noopener">e2e-cypress</a></li>
  16 |     </ul>


error: Line 14 exceeds the maximum line length of 120 (max-len) at src/components/HelloWorld.vue:14:1:
  12 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-typescript" target="_blank" rel="noopener">typescript</a></li>
  13 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint" target="_blank" rel="noopener">eslint</a></li>
> 14 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest" target="_blank" rel="noopener">unit-jest</a></li>
     | ^
  15 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-e2e-cypress" target="_blank" rel="noopener">e2e-cypress</a></li>
  16 |     </ul>
  17 |     <h3>Essential Links</h3>


error: Line 15 exceeds the maximum line length of 120 (max-len) at src/components/HelloWorld.vue:15:1:
  13 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint" target="_blank" rel="noopener">eslint</a></li>
  14 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest" target="_blank" rel="noopener">unit-jest</a></li>
> 15 |       <li><a href="https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-e2e-cypress" target="_blank" rel="noopener">e2e-cypress</a></li>
     | ^
  16 |     </ul>
  17 |     <h3>Essential Links</h3>
  18 |     <ul>


error: Line 29 exceeds the maximum line length of 120 (max-len) at src/components/HelloWorld.vue:29:1:
  27 |       <li><a href="https://router.vuejs.org" target="_blank" rel="noopener">vue-router</a></li>
  28 |       <li><a href="https://vuex.vuejs.org" target="_blank" rel="noopener">vuex</a></li>
> 29 |       <li><a href="https://github.com/vuejs/vue-devtools#vue-devtools" target="_blank" rel="noopener">vue-devtools</a></li>
     | ^
  30 |       <li><a href="https://vue-loader.vuejs.org" target="_blank" rel="noopener">vue-loader</a></li>
  31 |       <li><a href="https://github.com/vuejs/awesome-vue" target="_blank" rel="noopener">awesome-vue</a></li>
  32 |     </ul>


6 errors found.

Module Warning (from ./node_modules/eslint-loader/index.js):
error: Missing semicolon (semi) at src/components/HelloI18n.vue:11:56:
   9 | 
  10 | <script lang="ts">
> 11 | import { Component, Vue } from 'vue-property-decorator'
     |                                                        ^
  12 | 
  13 | @Component
  14 | export default class HelloI18n extends Vue {


1 error found.
1 error potentially fixable with the `--fix` option.

You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file.                                                                                                                                                                                    Type checking in progress...  

  App running at:
  - Local:   http://localhost:8080/ 
  - Network: http://10.0.0.182:8080/

  Note that the development build is not optimized.
  To create a production build, run npm run build.

No type errors found
Version: typescript 3.4.5
Time: 10798ms
```

**Review by faraonc — CHANGES_REQUESTED** [review]
Some npm and lint issues

File: package.json (hunk 12 above)
Comment (faraonc) [reply package.json:16]: I did hack this line `responseDeserializeFn: (bytes: Uint8Array) => Response);` just to try it out.

File: src/store/modules/gateway/types.ts [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import { AppGatewayServiceClient, grpc } from '../../types';
+
+export interface ModuleState {
+  grpcClient: AppGatewayServiceClient;
+  httpHeader: grpc.Metadata;
+}
```
Comment (faraonc) [inline src/store/modules/gateway/types.ts]: Can we just get rid of this file and move the interface to `state.ts` just like `ActionContext` in `actions.ts`? This is just an extra file.

File: src/store/modules/gateway/actions.ts (hunk 10 above)
Comment (faraonc) [inline src/store/modules/gateway/actions.ts]: We need to update `VUE_APP_HOST_NAME` in our slack channel config file

Comment (faraonc) [reply src/store/modules/gateway/actions.ts]: I think `||` and `!!` exist for reasons like this. I think this a language feature that we should take advantage.

Comment (faraonc) [reply src/store/modules/gateway/actions.ts]: Normally, there is another reviewer to settle differences. Maybe we can get opinion from @yeseulan? She worked at Disney as a frontend dev using Angular.

Comment (faraonc) [reply src/store/modules/gateway/actions.ts]: Are you using this value `http://localhost:50056`?

**Review by faraonc — CHANGES_REQUESTED** [review]
I suggest we find another reviewer to settle code review arguments.

File: tests/unit/store/gateway/mutations.spec.ts [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+  const { SET_HTTP_HEADER } = mutations;
+
+  it('should equal to payload', () => {
+    const hostname: string = 'test-address';
+    SET_HTTP_HEADER(state, hostname);
+    expect(state.authHeader).toEqual(hostname);
+  });
+});
+
+// https://github.com/grpc/grpc-web/pull/540
```
Comment (faraonc) [inline tests/unit/store/gateway/mutations.spec.ts:16]: We need a task to track this.

**Review by faraonc — CHANGES_REQUESTED** [review]

File: src/store/modules/gateway/actions.ts (hunk 10 above)
Comment (kimlisa) [reply src/store/modules/gateway/actions.ts]: Alright. You guys have industry level experience I don't.

File: src/store/modules/gateway/actions.ts (hunk 11 above)
Comment (kimlisa) [reply src/store/modules/gateway/actions.ts]: yes

File: src/store/modules/gateway/types.ts (hunk 13 above)
Comment (kimlisa) [reply src/store/modules/gateway/types.ts]: I meant to delete this file. It's already moved to root `types.ts`

File: src/store/modules/gateway/actions.ts (hunk 10 above)
Comment (kimlisa) [reply src/store/modules/gateway/actions.ts]: Updated. Yes its the envoy proxy port

File: tests/unit/store/gateway/mutations.spec.ts (hunk 14 above)
Comment (kimlisa) [reply tests/unit/store/gateway/mutations.spec.ts:16]: https://github.com/hwsc-org/hwsc-frontend/issues/27

File: package.json (hunk 12 above)
Comment (kimlisa) [reply package.json:16]: I was going to fix this on another branch. But I fixed it anyways. The only lint errors now are `(bytes: Uint8Array)` related.

**Review by faraonc — APPROVED** [review]
thank you and great work. TU
